colossus/src/main/scala/colossus/core/IOSystem.scala

```diff
@@ -15,42 +16,70 @@ import scala.concurrent.{ExecutionContext, Future}
 
 object IOSystem {
 
-  def apply(config: IOSystemConfig, metrics: MetricSystem)(implicit system: ActorSystem): IOSystem = {
-    val workerManager = system.actorOf(Props(classOf[WorkerManager], WorkerManagerConfig(config.numWorkers, metrics)), name = s"${config.name}-manager")
-    val sys = IOSystem(workerManager, config, metrics, system)
-    workerManager ! WorkerManager.Initialize(sys)
-    sys
-  }
-
-  def apply(name: String, numWorkers: Int, metrics: MetricSystem)
-    (implicit system: ActorSystem): IOSystem = apply(IOSystemConfig(name, numWorkers), metrics)
-
-
-  def apply(name: String, numWorkers: Int)(implicit system: ActorSystem): IOSystem = apply(name, numWorkers, MetricSystem.deadSystem)
+  val ConfigRoot = "colossus.io-system"
 
-  def apply(name: String, metrics: MetricSystem)(implicit  system: ActorSystem): IOSystem = {
-    val numWorkers = Runtime.getRuntime.availableProcessors()
-    apply(name, numWorkers, metrics)
+  /**
+    * Create a new IOSystem, using only the defaults provided by the corresponding "colossus.io-system" config path.
+    * A Config object will be created via {{{ConfigFactory.load()}}}
+    * @param sys
+    * @return
+    */
+  def apply()(implicit sys : ActorSystem) : IOSystem = {
+    apply(ConfigRoot)
   }
 
-  def apply(name: String = "iosystem", numWorkers: Option[Int] = None)(implicit sys: ActorSystem = ActorSystem(name)): IOSystem = {
-    val workers = numWorkers.getOrElse (Runtime.getRuntime.availableProcessors())
-    val metrics = MetricSystem(MetricAddress.Root / name)
-    apply(name, workers, metrics)
+  /**
+    * Create a new IOSystem using the supplied configPath.  This configPath will be overlaid on top of the default "colossus.io-system"
+    * config path.
+    * A Config object will be created via {{{ConfigFactory.load()}}}
+    *
+    * @param configPath  The path to the configuration.
+    * @param sys
+    * @return
+    */
+  def apply(configPath : String)(implicit sys : ActorSystem) : IOSystem = {
+    apply(configPath, ConfigFactory.load())
   }
 
-}
+  /**
+    * Create a new IOSystem by loading its config from the specified configPath.  This configPath will be overlaid on top of the default "colossus.io-system" config path.
+    * @param configPath The path to the configuration
+    * @param config The Config source to query
+    * @param sys
+    * @return
+    */
```

Comment:
If push comes to shove, L54 might be pushed into some utility object.  I'm holding off on that until I do more integration with the Server and Clients.